lib/AST/ASTImporter.cpp

```diff
@@ -6081,6 +6082,31 @@ Expr *ASTNodeImporter::VisitUnaryExprOrTypeTraitExpr(
                                           Importer.Import(E->getRParenLoc()));
 }
 
+Expr *ASTNodeImporter::VisitTypeTraitExpr(TypeTraitExpr *E) {
+  QualType T = Importer.Import(E->getType());
+  if (T.isNull())
+    return nullptr;
+
+  unsigned NumArgs = E->getNumArgs();
+  llvm::SmallVector<TypeSourceInfo *, 2> ToArgTypes(NumArgs);
+  for (unsigned AI = 0, AE = NumArgs; AI != AE; ++AI) {
+    TypeSourceInfo *FromArgType = E->getArg(AI);
+    TypeSourceInfo *ToArgType = Importer.Import(FromArgType);
+    if (!ToArgType)
+      return nullptr;
+    ToArgTypes[AI] = ToArgType;
+  }
+  TypeSourceInfo **ToArgTypesArray = new (Importer.getToContext())
+      TypeSourceInfo*[NumArgs];
+  for (unsigned AI = 0, AE = NumArgs; AI != AE; ++AI)
```

Comment:
Why do we need this 2. for loop?
Why not define the Array at the beginning and fill it with elements in the first loop?